src/__tests__/dsl.test.ts

```diff
@@ -57,5 +65,39 @@ describe('parseDSL', () => {
         let sx = parseDSL('payload contains "something"');
         expect(sx.lexErrors).toHaveLength(0);
         expect(sx.parseErrors).toHaveLength(0);
+        let payload = Buffer.from('something').toString('base64');
+        expect(
+            filterEvent(generateTestEvent(445, '123', '192.168.1.1', payload), sx.cst),
+        ).toBeTruthy();
+    });
+
+    test('parsing payload ne', () => {
+        let sx = parseDSL('not payload contains "banana"');
+        expect(sx.lexErrors).toHaveLength(0);
+        expect(sx.parseErrors).toHaveLength(0);
+        let payload = Buffer.from('something').toString('base64');
+        expect(
+            filterEvent(generateTestEvent(445, '123', '192.168.1.1', payload), sx.cst),
+        ).toBeTruthy();
+    });
+
+    test('payload contains and tcp.port', () => {
+        let sx = parseDSL('payload contains "something" and tcp.port == 445');
+        expect(sx.lexErrors).toHaveLength(0);
+        expect(sx.parseErrors).toHaveLength(0);
+        let payload = Buffer.from('something').toString('base64');
+        expect(
+            filterEvent(generateTestEvent(445, '123', '192.168.1.1', payload), sx.cst),
+        ).toBeTruthy();
+    });
+
+    test('parsing payload ne and tcp.port eq', () => {
+        let sx = parseDSL('not payload contains "banana" and tcp.port != 445');
+        expect(sx.lexErrors).toHaveLength(0);
+        expect(sx.parseErrors).toHaveLength(0);
+        let payload = Buffer.from('something').toString('base64');
+        expect(
+            filterEvent(generateTestEvent(445, '123', '192.168.1.1', payload), sx.cst),
+        ).toBeTruthy();
```

Comment:
@dkumiszhan shouldn't this be false here? TRUE AND FALSE = FALSE

Reply:
Yes, there was a bug which I am attempting to fix in https://github.com/honeynet/ochi/pull/69